### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'

### ISSUE THREAD

**Issue by ulrichard** [libkeepass/pykeepass#319]
We have used pykeepass for a while. Now I wanted to use the same code unchanged in a different environment. Now I run into the following error:
```
#13 4.694 filename = PosixPath('keepass/hsm_secrets_db.kdbx'), transformed_key = None
#13 4.694 
#13 4.694     def save(self, filename=None, transformed_key=None):
#13 4.694         """Save current database object to disk.
#13 4.694     
#13 4.694         Args:
#13 4.694             filename (:obj:`str`, optional): path to database or stream object.
#13 4.694                 If None, the path given when the database was opened is used.
#13 4.694                 PyKeePass.filename is unchanged.
#13 4.694             transformed_key (:obj:`bytes`, optional): precomputed transformed
#13 4.694                 key.
#13 4.694         """
#13 4.694     
#13 4.694         if not filename:
#13 4.694             filename = self.filename
#13 4.694     
#13 4.694         if hasattr(filename, "write"):
#13 4.694             KDBX.build_stream(
#13 4.694                 self.kdbx,
#13 4.694                 filename,
#13 4.694                 password=self.password,
#13 4.694                 keyfile=self.keyfile,
#13 4.694                 transformed_key=transformed_key
#13 4.694             )
#13 4.694         else:
#13 4.694             # save to temporary file to prevent database clobbering
#13 4.694             # see issues 223, 101
#13 4.694             # FIXME python2 - use pathlib.Path.withsuffix
#13 4.694 >           filename_tmp = filename + '.tmp'
#13 4.694 E           TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'
```
That is line 162 of https://github.com/libkeepass/pykeepass/blob/master/pykeepass/pykeepass.py

Depending on the version of pykeepass (3.2.1-2 from debian  or  4.0.2 from pip) I use, the error happens when running our unit tests, or during the execution of our scripts.
I am puzzled as to why this shows up only now.

**Comment by Evidlo** [comment]
This should be fixed in https://github.com/libkeepass/pykeepass/commit/817450b87fd2fbe062d409ee3213805e8e119265 to be released in version 4.0.4.

**Comment by ulrichard** [comment]
Thanks for fixing this. In the meantime I also found out that version 4.0.1 (which is in the Alpine repo) was unaffected by this.
